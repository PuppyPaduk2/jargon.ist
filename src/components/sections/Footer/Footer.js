// @flow

import React from 'react';
import Link from 'gatsby-link';

import { Container } from '../../ui';

import Logo from '../Logo';

import './Footer.scss';

const Footer = () => (
  <footer className="c-footer">
    <Container>
      <div className="c-footer__inner">
        <div>
          <Logo className="u-color-white" />
          <small>MIT lisansı ile geliştirilmiştir.</small>
        </div>

        <div>
          <h4 className="u-text-uppercase u-clear-gap-top u-gap-bottom-xsmall u-color-muted">
            Jargon
          </h4>
          <ul className="c-footer__menu">
            <li className="c-footer__menu-item">
              <Link to="/" className="c-footer__menu-link">
                Ana sayfa
              </Link>
            </li>
            <li className="c-footer__menu-item">
              <Link to="/dizin" className="c-footer__menu-link">
                Dizin
              </Link>
            </li>
            <li className="c-footer__menu-item">
              <Link to="/konular" className="c-footer__menu-link">
                Konular
              </Link>
            </li>
          </ul>
        </div>
        <div>
          <h4
            className="u-text-uppercase u-clear-gap-top u-gap-bottom-xsmall u-color-muted"
            lang="en"
          >
            Github
          </h4>
          <ul className="c-footer__menu">
            <li className="c-footer__menu-item">
              <a
                href={process.env.GATSBY_REPO_URL}
                className="c-footer__menu-link"
                target="_blank"
                rel="noopener noreferrer"
              >
                Kaynağı Görüntüle
              </a>
            </li>
            <li className="c-footer__menu-item">
              <a
                href={process.env.GATSBY_REPO_URL}
                className="c-footer__menu-link"
                target="_blank"
                rel="noopener noreferrer"
              >
                Katkıda Bulun
              </a>
            </li>
            <li className="c-footer__menu-item">
              <a
                href={`${process.env.GATSBY_REPO_URL || ''}/pulls?q=is:pr+is:open+label:new-jargon`}
                className="c-footer__menu-link"
                target="_blank"
                rel="noopener noreferrer"
              >
                Onay Bekleyen Jargonlar
              </a>
            </li>
            <li className="c-footer__menu-item">
              <a
                href={`${process.env.GATSBY_REPO_URL || ''}/issues/new`}
                className="c-footer__menu-link"
                target="_blank"
                rel="noopener noreferrer"
              >
                Sorun Bildir
              </a>
            </li>
            <li className="c-footer__menu-item">
              <a
                href={`${process.env.GATSBY_REPO_URL || ''}#katkida-bulunanlar`}
                className="c-footer__menu-link"
                target="_blank"
                rel="noopener noreferrer"
              >
                Katkıda Bulunanlar
              </a>
            </li>
          </ul>
        </div>
      </div>
    </Container>
  </footer>
);

export default Footer;
