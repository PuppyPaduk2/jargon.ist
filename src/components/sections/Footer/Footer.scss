@import '../../../styles/abstracts/index';

.c-footer {
  background-color: $secondary-color;
  color: $color-white;
  flex-shrink: 0;
  padding: pad(large) 0;

  &__inner {
    display: grid;
    grid-gap: px-to-rem(32px);
    grid-template-columns: 3fr 1fr 1fr;
  }

  &__logos {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
  }

  &__logo {
    &:hover {
      color: $primary-color;
      text-decoration: none;
    }
  }

  &__menu {
    list-style: none;
    margin: 0;
    padding: 0;
  }

  &__menu-item {
    margin-bottom: gap(2xsmall);
  }

  &__menu-link {
    color: $color-white;
  }
}
