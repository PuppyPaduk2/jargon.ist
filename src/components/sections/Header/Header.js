// @flow

import React from 'react';
import Link from 'gatsby-link';

import { Container } from '../../ui';

import Logo from '../Logo';
import Menu from '../Menu';
import Search from '../Search';

import './Header.scss';

const Header = () => (
  <header className="c-header">
    <Container>
      <div className="c-header__inner">
        <Link to="/" className="c-header__logo">
          <Logo />
        </Link>

        <Search />

        <Menu />
      </div>
    </Container>
  </header>
);

export default Header;
