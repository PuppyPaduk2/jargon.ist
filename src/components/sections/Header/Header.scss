@import '../../../styles/abstracts/index';

.c-header {
  background-color: $secondary-color;
  flex-shrink: 0;

  &__inner {
    align-items: center;
    display: grid;
    grid-gap: px-to-rem(32px);
    grid-template-columns: 1fr 4fr 2fr;
    padding: px-to-rem(16rem) 0;
  }

  &__logo {
    &:hover {
      color: $primary-color;
      text-decoration: none;
    }
  }
}
