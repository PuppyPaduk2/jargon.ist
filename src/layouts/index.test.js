// @flow

import React from 'react';
import { shallow } from 'enzyme';
import Helmet from 'react-helmet';

import Layout from './';

import Header from '../components/sections/Header';
import Footer from '../components/sections/Footer';

it('renders correctly', () => {
  const wrapper = shallow(<Layout>{() => <h1>Site</h1>}</Layout>);

  expect(wrapper).toMatchSnapshot();
});

it('renders Header', () => {
  const wrapper = shallow(<Layout>{() => <h1>Site</h1>}</Layout>);

  expect(wrapper.find(Header)).toHaveLength(1);
});

it('renders Footer', () => {
  const wrapper = shallow(<Layout>{() => <h1>Site</h1>}</Layout>);

  expect(wrapper.find(Footer)).toHaveLength(1);
});

it('renders Helmet with titleTemplate', () => {
  const wrapper = shallow(<Layout>{() => <h1>Site</h1>}</Layout>);

  expect(wrapper.find(Helmet).prop('titleTemplate')).toContain('%s');
});
