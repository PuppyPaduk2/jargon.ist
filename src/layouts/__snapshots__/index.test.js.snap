// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<React.Fragment>
  <HelmetWrapper
    defaultTitle="jargon.ist"
    defer={true}
    encodeSpecialCharacters={true}
    meta={
      Array [
        Object {
          "content": "Jargon.ist, bilgisayar bilimleri alanında kullanılan bazı jargonların topluluk tarafından Türkçe olarak açıklanmasını sağlayan bir web uygulamasıdır.",
          "name": "description",
        },
        Object {
          "content": "Jargon.ist, bilgisayar bilimleri alanında kullanılan bazı jargonların topluluk tarafından Türkçe olarak açıklanmasını sağlayan bir web uygulamasıdır.",
          "name": "og:description",
        },
        Object {
          "content": "jargon.ist",
          "name": "og:title",
        },
        Object {
          "content": "https://jargon.ist/social.png",
          "name": "og:image",
        },
        Object {
          "content": "jargon.ist",
          "name": "og:site_name",
        },
        Object {
          "content": "summary",
          "name": "twitter:card",
        },
        Object {
          "content": "jargon.ist",
          "name": "twitter:title",
        },
        Object {
          "content": "https://jargon.ist/social.png",
          "name": "twitter:image",
        },
        Object {
          "content": "Jargon.ist, bilgisayar bilimleri alanında kullanılan bazı jargonların topluluk tarafından Türkçe olarak açıklanmasını sağlayan bir web uygulamasıdır.",
          "name": "twitter:description",
        },
      ]
    }
    titleTemplate="%s - jargon.ist"
  />
  <Header />
  <main
    className="o-main"
  >
    <h1>
      Site
    </h1>
  </main>
  <Footer />
</React.Fragment>
`;
